astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,374 @@
+
+    def __init__(self):
+        return
+
+    def get_file(self, url, filename=None, verbose=False):
+        if filename is None:
+            raise ValueError("filename must be specified")
+        urllib.request.urlretrieve(url, filename)
+        if "/" not in filename:
+            print("File " + str(filename) + " downloaded to current directory")
+        else:
+            print("File " + str(filename) + " downloaded")
+
+    def get_table(self, url, filename=None, output_format='votable',
+                  verbose=False):
+        if filename is None:
+            raise ValueError("filename must be specified")
+        urllib.request.urlretrieve(url, filename)
+        table = modelutils.read_results_table_from_file(filename,
+                                                        str(output_format))
+        return table
+
+
+Handler = ESAHubbleHandler()
+
+
+class ESAHubbleClass(object):
+
+    data_url = conf.DATA_ACTION
+    metadata_url = conf.METADATA_ACTION
+
+    def __init__(self, url_handler=None, tap_handler=None):
+        if url_handler is None:
+            self._handler = Handler
+        else:
+            self._handler = url_handler
+
+        if tap_handler is None:
+            self._tap = TapPlus(url="http://hst.esac.esa.int"
+                                    "/tap-server/tap/")
+        else:
+            self._tap = tap_handler
+
+    def get_product(self, observation_id, calibration_level="RAW",
+                    filename=None, verbose=False):
+        """ Download products from EHST
+
+            Parameters
+            ----------
+            observation_id : string, id of the observation to be downloaded,
+            mandatory
+            The identifier of the observation we want to retrieve, regardless
+            of whether it is simple or composite.
+            calibration_level : string, calibration level, optional, default
+            'RAW'
+            The identifier of the data reduction/processing applied to the
+            data. By default, the most scientifically relevant level will be
+            chosen. RAW, CALIBRATED, PRODUCT or AUXILIARY
+            filename : string, file name to be used to store the artifact,
+            optional, default None
+            File name for the observation.
+            verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+            Returns
+            -------
+            None. It downloads the observation indicated
+        """
+
+        obs_id = "OBSERVATION_ID=" + observation_id
+        cal_level = "CALIBRATION_LEVEL=" + calibration_level
+        link = self.data_url + obs_id + "&" + cal_level
+        if filename is None:
+            filename = observation_id + ".tar"
+        print(link)
+        return self._handler.get_file(link, filename, verbose)
+
+    def get_artifact(self, artifact_id, filename=None, verbose=False):
+        """ Download artifacts from EHST
```

Comment:
It would be helpful to describe what an artifact is, and perhaps link to the full documentation on artifacts